packages/components/src/date-time/time.js

```diff
@@ -117,8 +121,16 @@ export function TimePicker( { is12Hour, currentTime, onChange } ) {
 	}
 
 	function update( name, value ) {
```

Comment:
This function is passed as the `onUpdate` callback for the `dayFormat` onBlur behaviour, so unfortunately the change here introduces a regression when switching dates. To replicate, highlight the date, enter a new date and tab away from the field — for me, it's removing 12 from the date value (and causes one of the unit tests to fail).

![Kapture 2021-11-26 at 10 46 45](https://user-images.githubusercontent.com/14988353/143508506-c767bded-4e96-4779-91dc-701770b884e0.gif)

There is a similar issue with the Year field, too, as that also uses `update`, though we probably wouldn't notice it with the Month field, since there are no more than 12 months in a year 😀

Should we add another function `updateHours` that includes the logic added to `update` and then within `updateHours` call `update`? We would then pass `updateHours` to the `UpdateOnBlurAsIntegerField` around line `249`.

Reply:
I should have fixed the issue by following @ntsekouras 's [inspiration](https://github.com/WordPress/gutenberg/pull/36878#discussion_r757417821) — the `from12hTo24h` function is not only called when the 12 hours format is enabled, and the input's name is `hours`.